### ISSUE TITLE
Improve consistency of multi-valued set?

### ISSUE THREAD

**Issue by technomancy** [bakpakin/Fennel#16]
The current way to set multiple values at once is like: `(set a b c (values 1 2 3))`

It seems to me it would be more consistent to support it as `(set a 1 b 2 c 3)` to make it more similar to `let`.

Or another option would be `(set [a b c] [1 2 3])` to make them both support destructuring.

What do you think?

**Comment by bakpakin** [comment]
I think proper destructuring is the way to go. However, the current destructuring is overly complicated and needs improvement. In fixing issue #15, I have simplified the destructuring code. It should be easier to modify after #15 is merged.

There needs to be some way of properly handling multiple arguments. Multiple arguments in return values is an intrinsic feature in Lua that needs tightly integrated support. For example, many coroutine functions return two values, a status and optional error method. I would also prefer to not have
to pack/unpack return values to do this.

 I was thinking that in the destructuring, arrays [square brackets] and lists (parens) could have different semantics. Arrays would index into the first return value, while lists would index all return values.

Example:
```
(set many (fn [] (values [:a :b :c :d] 2 3 4)))

;; Prints [:a :b :c :d] 2 3 4
(let [(a b c d) (many)]
  (print a b c d)) 

;; Prints :a :b :c :d
(let [[a b c d] (many)]
  (print a b c d))
```

This would be flexible enough, but might introduce confusion. I'm open to other suggestions about handling multiple return values. I was curious so I looked at what common lisp does and found [this](https://www.cs.cmu.edu/Groups/AI/html/cltl/clm/node94.html). It doesn't look particularly elegant (multiple-values-bind), or well integrated into the core language, but perhaps its not that important in CL. Also, I'm not particularly familiar with common lisp.
